src/main/src/test/java/org/geoserver/security/decorators/DefaultSecureCatalogFactoryTest.java

```diff
@@ -0,0 +1,43 @@
+package org.geoserver.security.decorators;
+
+import org.easymock.EasyMock;
+import org.geoserver.security.WrapperPolicy;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class DefaultSecureCatalogFactoryTest {
```

Comment:
Test is not checking ``canSecure``

Reply:
Good point, I will add a new test for `canSecure` ..